nifi-commons/nifi-flow-encryptor/src/main/java/org/apache/nifi/flow/encryptor/StandardFlowEncryptor.java

```diff
@@ -49,38 +42,27 @@ public class StandardFlowEncryptor implements FlowEncryptor {
      */
     @Override
     public void processFlow(final InputStream inputStream, final OutputStream outputStream, final PropertyEncryptor inputEncryptor, final PropertyEncryptor outputEncryptor) {
-        try (final PrintWriter writer = new PrintWriter(new OutputStreamWriter(outputStream))) {
-            try (final BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
-                reader.lines().forEach(line -> {
-                    final Matcher matcher = ENCRYPTED_PATTERN.matcher(line);
-
-                    final StringBuffer sb = new StringBuffer();
-                    boolean matched = false;
-                    while (matcher.find()) {
-                        final String outputEncrypted = getOutputEncrypted(matcher.group(FIRST_GROUP), inputEncryptor, outputEncryptor);
-                        matcher.appendReplacement(sb, outputEncrypted);
-                        matched = true;
-                    }
-
-                    final String outputLine;
-                    if (matched) {
-                        matcher.appendTail(sb);
-                        outputLine = sb.toString();
-                    } else {
-                        outputLine = line;
-                    }
-
-                    writer.println(outputLine);
-                });
+        final BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
+        if (bufferedInputStream.markSupported()) {
+            final byte[] firstByte = new byte[1];
+            bufferedInputStream.mark(1);
+            try {
+                bufferedInputStream.read(firstByte, 0, 1);
```

Comment:
This could be changed to use the simple `read()` method:
```suggestion
            bufferedInputStream.mark(1);
            try {
                final int firstByte = bufferedInputStream.read();
```

Reply:
Making the changes